python/tvm/relay/expr.py

```diff
@@ -415,6 +415,27 @@ def realize(self):
         The corresponding normal expression.
         """
         return _expr.TempExprRealize(self)
+@register_relay_node
+class TypeHint(Expr):
```

Comment:
Is it really best for type hints to be exprs? I do not mean that it is a bad idea to have them, but what I wonder is whether it might be possible to have them as auxiliary information directly on an expr or variable instead of as an expr in their own right. It would be preferable to avoid introducing another form of expression if possible.